### PR TITLE
Add first blocked mechanism.

### PR DESCRIPTION
**Pull request by zaclegarssure**
## What?
Add a block user functionality.
Now you won't be able to see meetups with blocked users on the map (and the list) and you won't be
able to join them.

### CODE REVIEW

File: app/src/main/java/com/github/palFinderTeam/palfinder/ProfileViewModel.kt [hunk 1]
```diff
@@ -94,4 +109,30 @@ class ProfileViewModel @Inject constructor(
         }
     }
 
+    /**
+     * We handle the block/unblock logic here to avoid handling coroutines in
+     * the adapter. Notice that we need to fetch again the user who wants to follow,
+     * that's because we need to have the more up to date info.
+     */
+    fun block(userId: String, otherId: String) {
```
Comment (TheblueMan003) [inline app/src/main/java/com/github/palFinderTeam/palfinder/ProfileViewModel.kt:117]: nit-peek: you could merge the block and unblock into a block toggle and thus reduce the amount of dupplicated code. but again just nit-peeking

Comment (zaclegarssure) [reply app/src/main/java/com/github/palFinderTeam/palfinder/ProfileViewModel.kt:117]: it improve coverage

Comment (TheblueMan003) [inline app/src/main/java/com/github/palFinderTeam/palfinder/ProfileViewModel.kt:117]: Also from what I see you can still be followed by & follow people you blocked.

Comment (zaclegarssure) [reply app/src/main/java/com/github/palFinderTeam/palfinder/ProfileViewModel.kt:117]: Ah nice catch will try to fix, there are other issues actually with the way the blocking work (in term of user experience) I might have to fix them for next sprint.

**Review by pgilliar — APPROVED** [review]
Not showing meetup is not a good idea. It became hard to join, why not just show a little icon to indicate someone block is already in the meetup. And maybe add that people you blocked can't join the meetup you created. I approve but todo for next sprint

**Review by TheblueMan003 — APPROVED** [review]
Other than the few querks above, Looks good!

**Pull request merged**